Mesh_3/include/CGAL/Labeled_mesh_domain_3.h

```diff
@@ -604,7 +606,13 @@ class Labeled_mesh_domain_3
    *
    */
   template<typename CGAL_NP_TEMPLATE_PARAMETERS>
-  static Labeled_mesh_domain_3 create_labeled_image_mesh_domain(const CGAL::Image_3& image_, const CGAL_NP_CLASS& np = parameters::default_values())
+  static
+  std::conditional_t<
+    CGAL::parameters::is_default_parameter<CGAL_NP_CLASS, internal_np::detect_features_param_t>::value,
+    Labeled_mesh_domain_3,
+    Mesh_domain_with_polyline_features_3<Labeled_mesh_domain_3>
+  >
+  create_labeled_image_mesh_domain(const CGAL::Image_3& image_, const CGAL_NP_CLASS& np = parameters::default_values())
```

Comment:
I wonder if it might be safer to use `auto`, for the return type.

```suggestion
  auto
  create_labeled_image_mesh_domain(const CGAL::Image_3& image_, const CGAL_NP_CLASS& np = parameters::default_values())
```

Currently, the return type is copy-pasted (here and in the line starting with `using Domain_type = ` in the body. If those types happen to differ, but in a way that a conversion is possible, then the Named Return Value Optimization (NRVO), or the move semantic, will not apply, and the domain will be copied or converted. So, I would vote for `auto` here, for safety. And even if I find that particularly problematic for this usage, because the return type is far from being evident at the first reading of the code.

Reply:
Note I finally had to use `auto` in 98abdce to avoid implicit conversion that would remove the "with_features" wrapper. This was happening in the demo.